hoot-core/src/main/cpp/hoot/core/io/OsmApiDbChangesetWriter.cpp

```diff
@@ -0,0 +1,162 @@
+}
+
+OsmApiDbChangesetWriter::~OsmApiDbChangesetWriter()
+{
+  _db.close();
+}
+
+void OsmApiDbChangesetWriter::write(const QString sql)
+{
+  LOG_INFO("Executing changeset SQL queries against OSM API database...");
+
+  QString changesetInsertStatement;
+  QString elementSqlStatements = "";
+
+  const QStringList sqlParts = sql.split(";");
+  for (int i = 0; i < sqlParts.size(); i++)
+  {
+    const QString sqlStatement = sqlParts[i];
+    if (i == 0)
+    {
+      if (!sqlStatement.toUpper().startsWith("INSERT INTO CHANGESETS"))
+      {
+        throw HootException(
+          "The first SQL statement in a changeset SQL file must create a changeset.");
+      }
+      else
+      {
+        changesetInsertStatement = sqlStatement + ";";
+      }
+    }
+    else
+    {
+      elementSqlStatements += sqlStatement + ";";
+    }
+  }
+
+  if (elementSqlStatements.trimmed().isEmpty())
+  {
+    throw HootException("No element SQL statements in sql file");
+  }
+
+  _db.transaction();
+  //had problems here when trying to prepare these queries (should they be?); the changeset writing
+  //needs to be done before the element writing, hence the separate queries
+  _execNoPrepare(changesetInsertStatement);
+  _execNoPrepare(elementSqlStatements);
+  _db.commit();
+
+  LOG_INFO("Changeset SQL queries execute finished against OSM API database.");
+}
+
+void OsmApiDbChangesetWriter::write(QFile& changesetSqlFile)
+{
+  if (!changesetSqlFile.fileName().endsWith(".osc.sql"))
+  {
+    throw HootException("Invalid file type: " + changesetSqlFile.fileName());
+  }
+
+  if (changesetSqlFile.open(QIODevice::ReadOnly))
+  {
+    write(changesetSqlFile.readAll());
+    changesetSqlFile.close();
+  }
+  else
+  {
+    throw HootException("Unable to open changeset file: " + changesetSqlFile.fileName());
+  }
+}
+
+bool OsmApiDbChangesetWriter::conflictExistsInTarget(const QString boundsStr, const QString timeStr)
+{
+  LOG_INFO("Checking for OSM API DB conflicts for changesets within " << boundsStr << " and " <<
+           "created after " << timeStr << "...");
+
+  const Envelope bounds = GeometryUtils::envelopeFromConfigString(boundsStr);
+  LOG_VARD(bounds.toString());
+
+  const QDateTime time = QDateTime::fromString(timeStr, OsmApiDb::TIME_FORMAT);
+  LOG_VARD(time);
+  if (!time.isValid())
+  {
+    throw HootException(
+      "Invalid timestamp: " + time.toString() + ".  Should be of the form " + OsmApiDb::TIME_FORMAT);
+  }
+
+  shared_ptr<QSqlQuery> changesetItr = _db.getChangesetsCreatedAfterTime(timeStr);
+
+  Envelope allChangesetsBounds;
+  while (changesetItr->next())
+  {
+    shared_ptr<Envelope> changesetBounds(
+      new Envelope(changesetItr->value(0).toDouble(),
+                   changesetItr->value(1).toDouble(),
+                   changesetItr->value(2).toDouble(),
+                   changesetItr->value(3).toDouble()));
+    LOG_VARD(changesetBounds->toString());
+    allChangesetsBounds.expandToInclude(changesetBounds.get());
```

Comment:
very good.  will do that.

I'd like to get the intersection part happening in the sql query instead to avoid this part, but not sure if that can be done yet.